android/src/com/mapswithme/maps/MwmApplication.java

```diff
@@ -161,7 +161,43 @@ public String getApkPath()
     }
   }
 
-  public String getDataStoragePath()
+  private static boolean isMapStorage(String path)
+  {
+    final File dir = new File(path);
+    if (TextUtils.isEmpty(path) || !Utils.isDirectoryReadable(dir))
```

Comment:
Before refactoring we checked directory for writability. Now we check it for readability only. This can lead to issues when device with maps on SD was upgraded to KitKat. On this device maps directory on SD will be readable but not writable (that's why we perform kitkat migration - to move maps from read-only location to read-write location).